R/create.R

```diff
@@ -194,30 +193,36 @@ create_from_github <- function(repo_spec,
   }
 
   if (open) {
-    open_project(proj_get())
+    open_project(proj_get(), restore = old_project)
   }
+  invisible(TRUE)
+
 }
 
-open_project <- function(path, rstudio = NA) {
-  rproj_path <- rproj_path(path)
+## `rstudio` arg here is about whether to attempt a launch in RStudio
+## `rstudio` arg of `create_*()` functions is about whether to add .Rproj file
+open_project <- function(path, restore = NA, rstudio = NA) {
   if (is.na(rstudio)) {
-    rstudio <- !is.na(rproj_path)
+    rstudio <- is_rstudio_project(path)
   }
 
   if (rstudio && rstudioapi::hasFun("openProject")) {
-    done("Opening project in RStudio")
-    rstudioapi::openProject(rproj_path, newSession = TRUE)
-  } else {
-    setwd(path)
-    done("Changing working directory to {value(path)}")
+    done("Opening new project {value(path_file(path))} in RStudio")
+    rstudioapi::openProject(rproj_path(path), newSession = TRUE)
+    ## TODO: check this is correct on rstudio server / cloud
+    if (!is.na(restore)) {
+      proj_set(restore, force = TRUE)
+    }
+    return(invisible(TRUE))
   }
```

Comment:
(Only commenting because we've been discussing early returns a lot)

This doesn't feel quite right as an early return to me. It would feel better as true and false branches of an if statement, because it's more symmetric: either open in RStudio or change the working directory.